### PR TITLE
TTV: GUI- extract import functionality

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
Extract the import functionality to be used in the TTV wizard.

Closes: #3548

### CODE REVIEW

**Review by eden-ohana — APPROVED** [review]
LGTM

File: webui/src/pages/repositories/repository/objects.jsx [hunk 1]
```diff
@@ -107,52 +101,19 @@ const ImportButton = ({ config, repo, reference, path, onDone, onClick, variant
...
-            onDone()
+            await runImport(updateStateFromImport,
+                `${destRef.current.value}`,
+                commitMsgRef.current.value,
+                sourceRef.current.value,
+                importBranch,
+                repo.id,
+                reference.id
+            );
+            onDone();
```
Comment (arielshaqed) [inline webui/src/pages/repositories/repository/objects.jsx:116]: (Not introduced in this PR but possibly relevant...)

Is `onDone` a synchronous call, or should this have been `await onDone()`?  In particular, if it's asynchronous, then IIRC both this _and the previous version_ lose exceptions that may occur while it runs.

**Review by arielshaqed — APPROVED** [review]
Thanks!

File: webui/src/pages/repositories/repository/objects.jsx [hunk 2]
```diff
@@ -107,52 +101,19 @@ const ImportButton = ({ config, repo, reference, path, onDone, onClick, variant
...
-                    setImportState({...initialState, error})
-                    throw error
-                }
-            }
-
-            await commits.commit(repo.id, importBranchResp.id, commitMsg, {}, metarange.id)
-            setImportState({inProgress: false, error: null, done: true, numObj: sum, isSourceValid: importState.isSourceValid})
-            onDone()
+            await runImport(updateStateFromImport,
+                `${destRef.current.value}`,
```
Comment (arielshaqed) [inline webui/src/pages/repositories/repository/objects.jsx:109]:
```suggestion
                destRef.current.value,
```

File: webui/src/pages/repositories/repository/objects.jsx [hunk 3]
```diff
@@ -107,52 +101,19 @@ const ImportButton = ({ config, repo, reference, path, onDone, onClick, variant
...
-                } else {
-                    setImportState({...initialState, error})
-                    throw error
-                }
-            }
-
-            await commits.commit(repo.id, importBranchResp.id, commitMsg, {}, metarange.id)
-            setImportState({inProgress: false, error: null, done: true, numObj: sum, isSourceValid: importState.isSourceValid})
-            onDone()
+            await runImport(updateStateFromImport,
```
Comment (arielshaqed) [inline webui/src/pages/repositories/repository/objects.jsx:108]: What happens if I reload during an import?

**Review by arielshaqed — COMMENTED** [review]
Thanks!...

File: webui/src/pages/repositories/repository/objects.jsx (hunk 3 above)
Comment (Jonathan-Rosenberg) [reply webui/src/pages/repositories/repository/objects.jsx:108]: then only the ranges that were created in the previous calls will exist. If no metarange was created, those ranges are gone.

File: webui/src/pages/repositories/repository/objects.jsx (hunk 1 above)
Comment (Jonathan-Rosenberg) [reply webui/src/pages/repositories/repository/objects.jsx:116]: `onDone` is a method that's passed from outside.. Don't think we should enforce a wait on it.
Nonetheless, the passed `onDone` function is just a state update function (toggles a boolean)

**Pull request merged**